### ISSUE TITLE
Tcp port not release after  deploy new struct of flows！

### ISSUE THREAD

**Issue by zhouyangwen** [tiagordc/node-red-contrib-tcp-server#2]
HI，i found the tcp server's port not be released after deploy new struct of flows，we should release the  tcp server on node.on("close", function()),just like below
>        
        node.on("close",function() {
            node.warn("node on close into");
            if (server) {
                node.warn("node on closeing");
                for (var connId in connectionPool) {
                    var socket = connectionPool[connId].socket;
                    socket.end();
                    socket.destroy();
                    socket.unref();
                }
                connectionPool = {};
                server.close();
            }
            node.warn("node on closed");
            node.status({});
        });

**Comment by ozdeadmeat** [comment]
yeah, this is probably why the TCP Client he made is also crashing.

https://github.com/tiagordc/node-red-contrib-tcp-client/issues/11

**Comment by jcovington16** [comment]
So this in theory would exit peacefully? Or will it still crash when this error occurs?

**Comment by TentiGianluca** [comment]
Hi,

the node don't release the port after a new deploy, your bugfix is already in production or not ? In the last case, ho i can modify and recompile the node ?

thank's

Gian-Luca

**Comment by TentiGianluca** [comment]
Hi, i've modified your code on my pc and the bug fix works, can you modify the published package with the change ?

thank's

Gian-Luca
